api/src/main/java/com/datatorrent/api/Launcher.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.api;
+
+import java.util.Iterator;
+import java.util.ServiceLoader;
+
+import org.apache.hadoop.conf.Configuration;
+
+/**
+ *
+ */
+public abstract class Launcher<H extends Launcher.AppHandle>
+{
+
+  public enum LaunchMode
+  {
+    LOCAL, CLUSTER
```

Comment:
The CLUSTER one could remain as is in case we add support for other distributed systems such as mesos. The factory can return the right *AppLauncher based on the environment in CLUSTER mode? What do you think?